### ISSUE TITLE
Deprecation warning in angular 4

### ISSUE THREAD

**Issue by dzonatan** [shlomiassaf/angular2-modal#323]
In angular 4 there is a deprecation warning:

```
The <template> element is deprecated. 
Use <ng-template> instead ("[WARNING ->]<template #innerView></template>
<template #template let-ctx>
     <template [swapCmp]="ctx.component""): ng:///ModalModule/ModalOverlay.html@0:0
```

**Comment by achimha** [comment]
If the module is changed to use `<ng-template>`, it will no longer work with Angular 2.x. This is a bit of an issue.

**Comment by dzonatan** [comment]
One day this warning will become error so it will no longer work with Angular _(probably)_ 5.x. 
So that's an issue either.

**Comment by achimha** [comment]
Pretty much all Angular 2+ projects are facing this issue. The question is, when is the time to stop supporting Angular 2.x in favor of silencing this warning?

I would argue that today is definitely not the time.

**Comment by doxavore** [comment]
Perhaps I'm missing something... how is anyone running this against Angular 4? The project seems to pin to `angular@^2.4.1`.

**Comment by achimha** [comment]
The Angular dependencies should actually be moved to `peerDependencies`. Still, it's not a fatal error with npm.

**Comment by rolandoldengarm** [comment]
npm shrink fails because of the unmet peerdependencies...

**Comment by williamBurgeson** [comment]
Ultimately, the "correct" way to proceed, IMO, would be to rename the project "angular-modal", as it is now established that Angular 2, 4 etc onwards are called simply Angular, as opposed to 1.x, which will be referred to as "AngularJS" going forward. Then we could have multiple versions, preferably tied to the versions of Angular - so v2 of this project would work with Angular 2, v4 will work with Angular 4, and this will also get round the issue of the caret in npm auto-upgrading and breaking everything...

**Comment by achimha** [comment]
I agree regarding dropping the "angular2" bit. However, does such a small difference (which is just a warning for now) really warrant having a separate repository?

I'm not a big fan of how the Angular team handled this `<template>` change in 4.0.

**Comment by rolandoldengarm** [comment]
I am against having multiple packages. People can still use the old version of angular2-modal if they want to stick at Angular 2.
If they want a new feature in angular2-modal, this may mean that they have to upgrade to Angular 4. Should never be a big deal...
Most of other packages we use just upgrade the peerdeps to ng4, or remove them completely.

**Comment by williamBurgeson** [comment]
To all intents and purposes ng2 was a completely new technology from ng1, however given we can't stay on ng2 forever it would be silly to refer to ng4, ng5 etc explicitly by the version number. In due course ng1 will fade into the past as a legacy technology, and we won't need to explicitly refer to the version number. 
Although these discussions around naming conventions can seem like a waste of time, it's exactly the same principle as within a project as renaming a variable from "dataService" to "weatherDataService" when we're about to introduce a non-weather data service.
"angular2-modal" was the correct name when the technology was called "angular 2". Now it's called "angular" a change of name is called for.
Angular 2 is dead; long live Angular!

**Comment by shlomiassaf** [comment]
fixed in 3.0.0
